engine/src/main/java/org/destinationsol/ui/SolUiSlider.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.ui;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.g2d.TextureAtlas;
+import com.badlogic.gdx.math.Rectangle;
+import com.badlogic.gdx.math.Vector2;
+import org.destinationsol.assets.Assets;
+import org.destinationsol.common.SolColor;
+
+public class SolUiSlider {
+    private static final float HANDLE_SCALE = 0.0125f;
+
+    private static final TextureAtlas.AtlasRegion sliderTexture = Assets.getAtlasRegion("engine:ui/slider");
+    private static final TextureAtlas.AtlasRegion sliderMarkerTexture = Assets.getAtlasRegion("engine:ui/sliderMarker");
+
+    private final Rectangle rectangle;
+    private float value;
+    private String text;
+    private final int trimAt;
+
+    public SolUiSlider(Rectangle sliderRectangle, String text, float startingValue, int trimAt) {
+        rectangle = sliderRectangle;
+        this.text = text;
+        if (startingValue > 1.0f) {
```

Comment:
I think using Math.min()/Math.max() could provide more readibility here and in the following lines.

Reply:
`SolMath.clamp()`